tests/Doctrine/Tests/ORM/Functional/Ticket/GH10532Test.php

```diff
@@ -0,0 +1,186 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Functional\Ticket;
+
+use Doctrine\ORM\Mapping as ORM;
+use Doctrine\Tests\OrmFunctionalTestCase;
+
+class GH10532Test extends OrmFunctionalTestCase
+{
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->createSchemaForModels(
+            GH10532A::class,
+            GH10532B::class,
+            GH10532C::class,
+            GH10532X::class
+        );
+    }
+
+    public function tearDown(): void
```

Comment:
@greg0ire you recently mentioned `tearDown` is no good/rarely necessary?

In this case, it seems the tests fail without it, possibly because we need a particular sequence to drop tables?

Reply:
Without `tearDown()` , the tests fail because something else drops table, in a wrong order?

Reply:
I tried removing `tearDown()`, and tests fail because of the insert on line 78, which violates a unique constraint.

Reply:
Yes… https://github.com/doctrine/orm/actions/runs/5158576597/jobs/9292414242

Though I cannot fully reproduce it locally.

Maybe somehow the automatic tear down does not work and so there are leftovers in the DB when the next test runs?

Reply:
I don't think there is any cleanup being done aside from a few models, so yeah the DB must be full of clutter, and that should be OK.